File: queries/_test_fixtures/python/calls.py
"""Python module with various call patterns for AST fixture testing."""

import os
import sys
from pathlib import Path


def call_external():
    """Calls to external functions."""
    os.getcwd()
    len([1, 2, 3])
    Path("/tmp").exists()


class CallerService:
    """Service with internal and external calls."""

    def do_work(self):
        self.validate()
        self.process()
        call_external()

    def validate(self):
        pass

    def process(self):
        self.validate()

File: queries/_test_fixtures/javascript/calls.js
/**
 * JavaScript calls fixture for AST fixture testing.
 */

const path = require('path');
const fs = require('fs');

function externalCall() {
    path.join('/tmp', 'file.txt');
    fs.readFileSync('/tmp/file.txt');
    console.log("hello");
}

class CallerService {
    doWork() {
        this.validate();
        this.process();
        externalCall();
    }

    validate() {
        return true;
    }

    process() {
        this.validate();
    }
}
